src/components/ButtonDropdown/ButtonDropdown.tsx

```diff
@@ -0,0 +1,213 @@
+      if (el) el.focus();
+    }
+    document.addEventListener('mousedown', handleOnClickOutside, false); /* 1 */
+    document.addEventListener(
+      'touchstart',
+      handleOnClickOutside,
+      false,
+    ); /* 1 */
+    return () => {
+      document.removeEventListener(
+        'mousedown',
+        handleOnClickOutside,
+        false,
+      ); /* 1 */
+      document.removeEventListener(
+        'touchstart',
+        handleOnClickOutside,
+        false,
+      ); /* 1 */
+    };
+  });
+
+  /**
+   * Handle click outside function
+   * 1) If the event isn't inside to the dropdown button itself or the entire
+   * component, then close the panel
+   */
+  const handleOnClickOutside = (event: MouseEvent | TouchEvent) => {
+    if (ref.current && !ref.current.contains(event.target as HTMLElement)) {
+      closePanel(); /* 1 */
+    }
+  };
+
+  /**
+   * Close the panel
+   * 1) Set active state to false
+   * 2) Return the focus to the button that triggered the dropdown when closed
+   */
+  function closePanel() {
+    setIsActive(false); /* 1 */
+
+    setTimeout(() => {
+      if (isActiveVar && buttonRef.current) {
+        buttonRef.current.focus(); /* 2 */
+      }
+    }, 1);
+  }
+
+  /**
+   * Handle keydown function
+   * 1) If the escape key is struck, close the panel
+   */
+  function handleKeyDown(e: React.KeyboardEvent) {
+    if (e.key === ESCAPE_KEYCODE) {
+      closePanel();
+    }
+  }
+
+  /**
+   * Toggle accordion panel
+   */
+  function togglePanel() {
+    setIsActive(!isActiveVar);
+  }
+
+  const dropdownMenuTriggerWithProps = React.Children.map(
+    dropdownMenuTrigger,
+    // TODO: improve `any` type
+    (child: any, i: number) => {
+      // Checking isValidElement is the safe way and avoids a typescript
+      // error too.
+      if (React.isValidElement(child)) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-expect-error TODO: fix "No overload matches this call" error
+        return React.cloneElement<Props>(child, {
+          ref: buttonRef,
+          onClick: togglePanel,
+          'aria-expanded': isActiveVar,
+        });
+      }
+    },
+  );
+
+  const handleBlur = (event: FocusEvent<HTMLDivElement>) => {
+    if (!event.currentTarget.contains(event.relatedTarget as HTMLElement)) {
+      if (isActiveVar) {
+        closePanel();
+      }
+    }
+  };
+
+  const componentClassName = clsx(
+    styles['button-dropdown'],
+    className,
+    isActiveVar && styles['eds-is-active'],
+    position === 'top-left' && styles['button-dropdown--top-left'],
+    position === 'top-right' && styles['button-dropdown--top-right'],
+    position === 'bottom-right' && styles['button-dropdown--bottom-right'],
+  );
+  return (
+    <div
+      className={componentClassName}
+      onBlur={handleBlur}
+      onKeyDown={(e) => handleKeyDown(e)}
+      ref={ref}
+      role="presentation"
```

Comment:
Can you help me understand the `role="presentation"` bits in this PR? What does this sound like to a screen reader?

Reply:
@dierat This was yelling at me that I needed a role on it since it's not an element that normally has a `keydown` associated with it. So I added `role="presentation"` since this really doesn't have any significance other than being able to know where a user can click outside. Don't think screenreaders need to "see" this div since it's mostly a grouping of things. The `aria-expanded` on the button should take care of the accessibility part of it.